### PR TITLE
chore: rate limit requests

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Rate limit the export dashboard endpoint.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/config.py [hunk 1]
```diff
@@ -264,6 +264,12 @@ def _try_json_readsha(filepath: str, length: int) -> str | None:
 # Configuration for scheduling queries from SQL Lab.
 SCHEDULED_QUERIES: dict[str, Any] = {}
 
+# Rate limiting
+RATELIMIT_ENABLED = True
+AUTH_RATE_LIMITED = True
+AUTH_RATE_LIMIT = "2 per 5 second"
+RATELIMIT_APPLICATION = "50 per second"
```
Comment (betodealmeida) [inline superset/config.py:271]: @dpgaspar I know these were introduced in FAB 4.3.0. Do you have an idea of what sensible defaults would be for Superset?

File: superset/models/dashboard.py [hunk 2]
```diff
@@ -382,7 +382,7 @@ def export_dashboards(  # pylint: disable=too-many-locals
     ) -> str:
         copied_dashboards = []
         datasource_ids = set()
-        for dashboard_id in dashboard_ids:
+        for dashboard_id in set(dashboard_ids):
```
Comment (john-bodley) [inline superset/models/dashboard.py:385]: If there's an issue with duplicates—where order doesn't matter—is there merit in changing the `export_dashboards` method to have `dashboard_ids` as `set[int]`?

File: superset/config.py [hunk 3]
```diff
@@ -264,6 +264,12 @@ def _try_json_readsha(filepath: str, length: int) -> str | None:
 # Configuration for scheduling queries from SQL Lab.
 SCHEDULED_QUERIES: dict[str, Any] = {}
 
+# Rate limiting
```
Comment (john-bodley) [inline superset/config.py:267]:
```suggestion
# FAB rate limiting
```

Maybe there's merit in mentioning FAB so it's clear where these limits are being enforced.

File: superset/config.py (hunk 1 above)
Comment (dpgaspar) [reply superset/config.py:271]: Airflow implemented the same, their using `auth_rate_limit = 5 per 40 second`. Not sure about rate limiting the entire application this way, rate limits are better when tied to a user and state is set globally, this is possible with [flask-limiter](https://flask-limiter.readthedocs.io/en/stable/) (what FAB is using for this) but needs further configuration.

Comment (betodealmeida) [reply superset/config.py:271]: Wait, I though these configuration keys would enable `flask-limiter` in FAB? That's not the case? How do we do that then?

File: superset/models/dashboard.py (hunk 2 above)
Comment (betodealmeida) [reply superset/models/dashboard.py:385]: Good point, let me fix it.

**Review by john-bodley — APPROVED** [review]

File: superset/config.py (hunk 1 above)
Comment (dpgaspar) [reply superset/config.py:271]: it's done exactly like that

**Review by dpgaspar — COMMENTED** [review]
2 per 5 seconds could be an harsh limit. Would be nice to add some docs, and reference `RATELIMIT_STORAGE_URI` and `RATELIMIT_REQUEST_IDENTIFIER` since as is flask-limiter is too simplistic. It's way better to limit an entire application by user request

**Pull request merged**